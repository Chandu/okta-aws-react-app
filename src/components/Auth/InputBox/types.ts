import { ChangeEvent } from 'react';

export type Props = {
    type: string;
    name?: string;
    value: string;
    label: string;
    errorText?: string;
    fullWidth?: boolean;
    placeholderText: string;
    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
};
