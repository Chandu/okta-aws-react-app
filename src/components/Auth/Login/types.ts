export type Props = {
    toggleWidget: () => void;
};
