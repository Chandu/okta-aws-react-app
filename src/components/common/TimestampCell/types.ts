import { Table } from '@medly-components/core';
import { GetComponentProps } from '@medly-components/utils';

export type ComponentType = GetComponentProps<typeof Table>['columns'][0]['component'];
