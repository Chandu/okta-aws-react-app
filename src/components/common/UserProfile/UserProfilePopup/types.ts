export interface Props {
    signOut: () => void;
    userName: string;
    showSeparator: boolean;
}
