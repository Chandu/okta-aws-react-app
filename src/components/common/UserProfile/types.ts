export interface Props {
    showSeparator?: boolean;
}
