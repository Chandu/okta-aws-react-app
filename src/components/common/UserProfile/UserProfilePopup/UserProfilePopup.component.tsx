import { Box, Button, Popover } from '@medly-components/core';
import { primaryTheme } from '@theme/primary';
import { getNameInitials } from '@utils';
import { FC, memo, useContext, useMemo } from 'react';
import { Props } from './types';
import { Overlay, Popup, PopupContent, StyledAvatar, UserInfo, UserTitle } from './UserProfilePopup.styled';

export const UserProfilePopup: FC<Props> = memo(({ signOut, userName }) => {
    const userNameInitials = useMemo(() => getNameInitials(userName), [userName]),
        [isPopoverVisible] = useContext(Popover.Context);

    return (
        <>
            {isPopoverVisible && <Overlay />}
            <StyledAvatar
                hoverBgColor={primaryTheme.colors.purple[400]}
                hoverTextColor={primaryTheme.colors.white}
                bgColor={isPopoverVisible ? primaryTheme.colors.purple[500] : ''}
                textColor={isPopoverVisible ? primaryTheme.colors.white : ''}
            >
                {userNameInitials}
            </StyledAvatar>
            <Popup placement="bottom" id="side-nav-popover">
                <PopupContent>
                    <UserInfo>
                        <StyledAvatar
                            leftMargin="0"
                            showSeparator={false}
                            bgColor={primaryTheme.colors.purple[500]}
                            textColor={primaryTheme.colors.white}
                            id="styledAvatar"
                        >
                            {userNameInitials}
                        </StyledAvatar>
                        <UserTitle textVariant="h5" uppercase>
                            {userName}
                        </UserTitle>
                    </UserInfo>
                    <Box>
                        <Button variant="solid" id="signOut" onClick={signOut} fullWidth>
                            Sign Out
                        </Button>
                    </Box>
                </PopupContent>
            </Popup>
        </>
    );
});

UserProfilePopup.displayName = 'UserProfilePopup';
