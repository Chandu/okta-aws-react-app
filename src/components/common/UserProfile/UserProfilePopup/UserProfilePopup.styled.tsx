import { Avatar, Popover, Text } from '@medly-components/core';
import { styled } from '@utils';

export const StyledAvatar = styled(Avatar)<{ leftMargin?: string; showSeparator?: boolean }>`
    z-index: 12;
    position: relative;
    cursor: pointer;
    transition: all 100ms ease-out;
    overflow: visible;
`;

export const UserTitle = styled(Text)`
    display: flex;
    flex-direction: column;
    text-align: left;
    color: ${({ theme }) => theme.colors.grey[900]};
    white-space: nowrap;
`;

export const Popup = styled(Popover.Popup)`
    width: fit-content;
    min-width: 23.2rem;
    top: 5.2rem;
    right: 0;
    background: #fff;
    box-shadow: 0 0.4rem 3.2rem rgba(96, 120, 144, 0.35);
    transform: none;
    left: auto;
    border-radius: 0.8rem;
    z-index: 12;
`;

export const PopupContent = styled('div')<{ active?: boolean }>`
    display: flex;
    flex-direction: column;

    & > * {
        cursor: pointer;
        padding: 2rem;
    }

    & > *:not(:last-child) {
        position: relative;
        &::after {
            content: '';
            position: absolute;
            left: 0;
            right: 0;
            width: 100%;
            height: 0.1rem;
            background: #dfe4e9;
            margin: auto;
            bottom: 0;
        }
    }

    & > *:not(:first-child) {
        &:hover {
            background-color: ${({ theme, active }) => !active && theme.colors.grey[50]};
        }

        &:active {
            background-color: ${({ theme, active }) => !active && theme.colors.grey[100]};
        }
    }

    & > :last-child {
        &:hover {
            border-radius: 0 0 0.8rem 0.8rem;
        }
    }
`;

export const UserInfo = styled('div')`
    display: flex;
    position: relative;
    align-items: center;
    flex-direction: row;
    padding-top: 2.4rem;
    & > ${StyledAvatar.Style} {
        margin-right: 1.2rem;
    }
`;

export const Overlay = styled('div')`
    transform: rotate(180deg);
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: transparent;
    z-index: 11;
`;

export const Container = styled('div')`
    display: flex;
    align-items: center;
    justify-content: space-around;
`;
