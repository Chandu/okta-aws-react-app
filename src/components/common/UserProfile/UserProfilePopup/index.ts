import { UserProfilePopup } from './UserProfilePopup.component';

export default UserProfilePopup;
