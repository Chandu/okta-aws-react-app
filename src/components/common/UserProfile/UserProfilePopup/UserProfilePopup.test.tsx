import { Popover } from '@medly-components/core';
import { fireEvent, render, screen } from '@test-utils';
import { oktaStorage } from '@utils/testData';
import { MemoryRouter } from 'react-router-dom';
import { Props } from './types';
import { UserProfilePopup } from './UserProfilePopup.component';

const signOutMock = jest.fn(),
    renderUserProfilePopup = ({
        showSeparator = false,
        userName = '',
        signOut = signOutMock,
        routerPath = '/sample'
    }: Partial<Props> & { routerPath?: string }) =>
        render(
            <MemoryRouter initialEntries={[routerPath]}>
                <Popover interactionType="click">
                    <UserProfilePopup {...{ showSeparator, userName, signOut }} />
                </Popover>
            </MemoryRouter>
        );

describe('UserProfilePopup', () => {
    beforeAll(() => {
        localStorage.setItem('okta-token-storage', JSON.stringify(oktaStorage));
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render properly', async () => {
        renderUserProfilePopup({ showSeparator: true, userName: 'Dummy' });
        const nameInitial = screen.getByText('D');
        expect(nameInitial).toBeInTheDocument();
        fireEvent.click(nameInitial);
        expect(await screen.findByText('Sign Out')).toBeInTheDocument();
    });

    it('should render user initials as empty string if username is not provided or undefined', async () => {
        const { container } = renderUserProfilePopup({ showSeparator: true, userName: undefined });
        expect(container).toMatchSnapshot();
    });
});
