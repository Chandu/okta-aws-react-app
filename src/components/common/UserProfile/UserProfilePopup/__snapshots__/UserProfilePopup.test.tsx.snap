// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfilePopup should render user initials as empty string if username is not provided or undefined 1`] = `
.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  text-align: initial;
  text-transform: uppercase;
  display: inline-block;
}

.c1 {
  display: inline-block;
  text-align: center;
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
  width: 4rem;
  height: 4rem;
  border-radius: 0.4rem;
  overflow: hidden;
  cursor: inherit;
  color: #6E14FF;
  background: #E9DCFF;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 .c3 {
  line-height: 4rem;
  font-size: 1.6rem;
  font-weight: 400;
  font-family: Poppins;
}

.c1 img {
  width: 4rem;
  height: 4rem;
  object-fit: cover;
  border: 0.1rem solid #dfe4e9;
  box-sizing: border-box;
  border-radius: 0.4rem;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  z-index: 12;
  position: relative;
  cursor: pointer;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  overflow: visible;
}

<div>
  <div
    class="c0"
    id="medly-popover-wrapper"
  >
    <div
      class="c1 c2"
    >
      <strong
        class="c3 c4"
      />
    </div>
  </div>
</div>
`;
